src/ast/node.ts

```diff
@@ -38,9 +39,15 @@ export default class Node implements AstType {
     this.type = type;
     this.tag = tag;
     this.annotations = [];
+    this.slots = {};
   }
 
   *walk(): Generator<Node, void, unknown> {
+    for (const slot of Object.values(this.slots)) {
```

Comment:
Would it be faster to make `this.slots` default to `undefined` and avoid the `Object.values(this.slots)` in the default case?

Reply:
I guess alternatively you could simplify this function by doing:
```js
    for (const child of this.children.concat(Object.values(this.slots))) {
      ...
```

or something similar

Reply:
I'm changing it to: `for (const child of [...Object.values(this.slots), ...this.children]) {`. I don't think it materially impacts performance either way.